src/analysis/normed_space/operator_norm.lean

```diff
@@ -16,13 +16,17 @@ import data.equiv.transfer_instance
 
 Define the operator norm on the space of continuous linear maps between normed spaces, and prove
 its basic properties. In particular, show that this space is itself a normed space.
+
+Since a lot of elementary properties don't require `∥x∥ = 0 → x = 0` we start setting up the
+theory for `semi_normed_space` and we specialize to `normed_space` when needed.
 -/
 
 noncomputable theory
 open_locale classical nnreal topological_space
 
-variables {𝕜 : Type*} {E : Type*} {F : Type*} {G : Type*}
-[normed_group E] [normed_group F] [normed_group G]
+variables {𝕜 : Type*} {E : Type*} {F : Type*} {G : Type*} {E₁ : Type*} {F₁ : Type*} {G₁ : Type*}
+[semi_normed_group E] [semi_normed_group F] [semi_normed_group G]
+[normed_group E₁] [normed_group F₁] [normed_group G₁]
```

Comment:
Hmm.. I'm not so keen on this approach, as is forces the reader to carefully inspect their variables to determine if a theorem is about a `semi_normed_group` or `normed_group`.

I would be much happier if you could sort the theorems into sections, and only have the relevant variables in each one.

(Actually, my first reading of this was "what is going on?! do you really need to have a `semi_normed_group` instance before you can have a `normed_group` instance? that's not what I remember", until a moment later I started noticing the subscripts.)

Reply:
Ah yes, that's much better! Let me do it.

Reply:
(Phew! I was worried I was dumping a lot of work on you there.)

Reply:
I reorganized the file: there is a section `semi_normed` first and then a section `normed`. Let's see if everything compiles... (it should be exactly the same as before, but maybe I messed up some namespace).

The diff is probably unreadable now, but I didn't change anything expect the order of the statements.